testgrid/config/jenkinsvalidate_test.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+    "testing"
+    "fmt"
+    "strings"
+    "path/filepath"
+    "io/ioutil"
+
+    "k8s.io/test-infra/testgrid/config/yaml2proto"
+    "github.com/golang/protobuf/proto"
+    "k8s.io/test-infra/testgrid/config/pb"
+)
+
+
+func TestJenkins(t *testing.T) {
+    // Jenkins Job Data should be gathered in test-infra/output
+    // If want to test locally, run jenkins-jobs test to collect all jobs first.
+
+    jenkinsjobs := make(map[string]bool)
+    files, err := filepath.Glob("../../output/*")
+    if err != nil {
+	t.Errorf("Failed to collect outputs")
+    }
+
+    for _, file := range files {
+        file = strings.TrimPrefix(file,"../../output/")
+	jenkinsjobs[file] = false
+    }
+
+    data, err := ioutil.ReadFile("config.yaml")
+    protobufData, err := yaml2proto.Yaml2Proto(data)
+
+    config := &config.Configuration{}
+    if err := proto.Unmarshal(protobufData, config); err != nil {
+        t.Errorf("Failed to parse config: %v\n", err)
+    }
+
+    // For now anything outsite k8s-jenkins/logs are considered to be fine
+    testgroups := make(map[string]bool)
+    for _, testgroup := range config.TestGroups {
+        if strings.Contains(testgroup.GcsPrefix, "kubernetes-jenkins/logs/"){
+            job := strings.TrimPrefix(testgroup.GcsPrefix,"kubernetes-jenkins/logs/")
+            testgroups[job] = false
+        }
+    }
+
+    // Cross check
+    // -- Each jenkins job need to have a match testgrid group
+    for jenkinsjob, _ := range jenkinsjobs {
+	if _, ok := testgroups[jenkinsjob]; ok {
+            testgroups[jenkinsjob] = true
+            jenkinsjobs[jenkinsjob] = true
+        }
+    }
+
+    // Conclusion
+    badjob := 0
+    for jenkinsjob, valid := range jenkinsjobs {
+        if !valid {
+            badjob++
+            fmt.Printf("Jenkins Job %v does not have a matching testgrid test\n", jenkinsjob)
+        }
+    }
+
+    badconfig := 0
+    for testgroup, valid := range testgroups {
+        if !valid {
+            badconfig++
+            fmt.Printf("Testgrid group %v does not have a matching jenkins job\n", testgroup)
+        }
+    }
+
+    if badconfig > 0 {
+        t.Errorf ("Total bad config - %v\n", badconfig)
```

Comment:
This will be more useful as a comma separated list of testgroups rather than an integer (and same for the next Errorf.

Take a look at https://golang.org/pkg/strings/#Join and https://tour.golang.org/moretypes/15